Doctrine/Orm/Filter/SearchFilter.php

```diff
@@ -59,10 +56,48 @@ public function __construct(
         PropertyAccessorInterface $propertyAccessor,
         array $properties = null
     ) {
-        $this->managerRegistry = $managerRegistry;
+        parent::__construct($managerRegistry, $properties);
+
         $this->iriConverter = $iriConverter;
         $this->propertyAccessor = $propertyAccessor;
-        $this->properties = $properties;
+    }
+
+    /**
+     * {@inheritdoc}
+     */
+    public function apply(ResourceInterface $resource, QueryBuilder $queryBuilder, Request $request)
+    {
+        $metadata = $this->getClassMetadata($resource);
+        $fieldNames = array_flip($metadata->getFieldNames());
+
+        foreach ($this->extractProperties($request) as $property => $value) {
+            if (!is_string($value) || !$this->isPropertyEnabled($property)) {
+                continue;
+            }
+
+            $partial = null !== $this->properties && self::STRATEGY_PARTIAL === $this->properties[$property];
+
+            if (isset($fieldNames[$property])) {
+                if ('id' === $property) {
```

Comment:
Don't you wish to put `'id'` as a constant too? This could be put in the `AbstractFilter`.